src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentServiceProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal interface IDocumentServiceProvider
+    {
+        /// <summary>
+        /// Gets a document specific service provided by the host identified by the service type. 
+        /// If the host does not provide the service, this method returns null.
+        /// </summary>
+        TService GetService<TService>() where TService : class, IDocumentService;
```

Comment:
why does TService need to be a class?

Reply:
don't know. @jasonmalinowski and @rynowak wanted it. for me, not sure whether that matter much.

Reply:
I guess a value-typed "service" seemed odd...

Reply:
I'm ok with this staying 'class'.  It just seemed odd to me to have this added.